packages/stacking/src/index.ts

```diff
@@ -0,0 +1,410 @@
+    amountMicroStx,
+    poxAddress,
+    cycles,
+    key,
+    burnBlockHeight,
+  }: LockStxOptions
+  ) {
+    const poxInfo = await this.getPoxInfo();
+    const contract = poxInfo.contract_id;
+
+    const txOptions = this.getLockTxOptions({
+      amountMicroStx,
+      cycles,
+      poxAddress,
+      contract,
+      burnBlockHeight,
+    });
+    const tx = await makeContractCall({
+      ...txOptions,
+      senderKey: key,
+    });
+
+    const res = await broadcastTransaction(tx, txOptions.network as StacksNetwork);
+    if (typeof res === 'string') {
+      return res;
+    }
+    throw new Error(`${res.error} - ${res.reason}`);
+  }
+
+  getLockTxOptions({
+    amountMicroStx,
+    poxAddress,
+    cycles,
+    contract,
+    burnBlockHeight,
+  }: {
+    cycles: number;
+    poxAddress: string;
+    amountMicroStx: BN;
+    contract: string;
+    burnBlockHeight: number;
+  }) {
+    const { hashMode, data } = convertBTCAddress(poxAddress);
+    const hashModeBuffer = bufferCV(new BN(hashMode, 10).toBuffer());
+    const hashbytes = bufferCV(data);
+    const address = tupleCV({
+      hashbytes,
+      version: hashModeBuffer,
+    });
+    const [contractAddress, contractName] = contract.split('.');
+    const network = this.network; 
+    const txOptions: ContractCallOptions = {
+      contractAddress,
+      contractName,
+      functionName: 'stack-stx',
+      // sum of uStx, address, burn_block_height, num_cycles
+      functionArgs: [
+        uintCV(amountMicroStx.toString(10)),
+        address,
+        uintCV(burnBlockHeight),
+        uintCV(cycles),
+      ],
+      validateWithAbi: true,
+      network,
+    };
+    return txOptions;
+  }
+
+  /**
+   * Check stacking status
+   * 
+   * @returns {Promise<StackerInfo>} that resolves to a StackerInfo object if the operation succeeds
+   */
+  async getStatus(): Promise<StackerInfo> {
+    const [contractAddress, contractName] = (await this.getPoxInfo()).contract_id.split('.');
+    const functionName = 'get-stacker-info';
+
+    return callReadOnlyFunction({
+      contractAddress,
+      contractName,
+      functionName,
+      senderAddress: this.address,
+      functionArgs: [
+        standardPrincipalCV(this.address)
+      ],
+      network: this.network
+    })
+    .then((responseCV: ClarityValue) => {
+      if (responseCV.type === ClarityType.OptionalSome) {
+        const someCV = responseCV as SomeCV;
+        const tupleCV: TupleCV = someCV.value as TupleCV;
+        const poxAddress: TupleCV = tupleCV.data["pox-addr"] as TupleCV;
+        const amountMicroStx: UIntCV = tupleCV.data["amount-ustx"] as UIntCV;
+        const firstRewardCycle: UIntCV = tupleCV.data["first-reward-cycle"] as UIntCV;
+        const lockPeriod: UIntCV = tupleCV.data["lock-period"] as UIntCV;
+        const version: BufferCV = poxAddress.data['version'] as BufferCV;
+        const hashbytes: BufferCV = poxAddress.data['hashbytes'] as BufferCV;
+      
+        return {
```

Comment:
would it be possible to return the txid/explorer URL for the stacking transaction that "reasoned" the status?